app/Makefile

```diff
@@ -33,15 +33,37 @@ else
 PY_RUN_CMD := docker-compose run $(DOCKER_EXEC_ARGS) --rm $(APP_NAME) poetry run
 endif
 
+# Docker user configuration
+#
+# Can be set by adding user=<username> and/ or uid=<id> to make command.
+#
+# If variables are not set explicitly: try looking up values from current
+# environment, otherwise fixed defaults.
+#
+# uid= defaults to 0 if user= set (which makes sense if user=root, otherwise you
+# probably want to set uid as well).
+#
+# Tested to work consistently on popular Linux flavors and Mac.
+ifeq ($(user),)
+RUN_USER ?= $(or $(strip $(USER)),nodummy)
+RUN_UID ?= $(or $(strip $(shell id -u)),4000)
+else
+RUN_USER = $(user)
+RUN_UID = $(or $(strip $(uid)),0)
+endif
+
+export RUN_USER
+export RUN_UID
+
 ##################################################
 # API Build & Run
 ##################################################
 
 build:
-	docker-compose build
+	docker-compose build $(APP_NAME)
```

Comment:
Doesn't this mean that `main-db` doesn't build?

Reply:
Kind of? In the `build` command, we're just building the image, and we aren't configuring the DB with a Dockerfile, so it doesn't really affect it one way or another if we have the specification for the image or not. It somewhat saves time because Docker doesn't need to check for configs for the DB